components/datetime/src/format/input.rs

```diff
@@ -17,24 +17,26 @@ use icu_time::{zone::UtcOffset, Time, TimeZone};
 // TODO(#2630) fix up imports to directly import from icu_calendar
 pub(crate) use icu_calendar::types::{DayOfMonth, MonthInfo, Weekday, YearInfo};
 
-#[derive(Debug, Copy, Clone)]
-pub(crate) struct ExtractedInput {
-    pub(crate) year: Option<YearInfo>,
-    pub(crate) month: Option<MonthInfo>,
-    pub(crate) day_of_month: Option<DayOfMonth>,
-    pub(crate) iso_weekday: Option<Weekday>,
-    pub(crate) day_of_year: Option<DayOfYearInfo>,
-    pub(crate) hour: Option<Hour>,
-    pub(crate) minute: Option<Minute>,
-    pub(crate) second: Option<Second>,
-    pub(crate) subsecond: Option<Nanosecond>,
-    pub(crate) time_zone_id: Option<TimeZone>,
-    pub(crate) offset: Option<UtcOffset>,
-    pub(crate) zone_variant: Option<TimeZoneVariant>,
-    pub(crate) local_time: Option<(Date<Iso>, Time)>,
+/// An input bag with all possible datetime input fields.
+#[derive(Debug, Copy, Clone, Default)]
+#[non_exhaustive]
+pub struct DateTimeInputUnchecked {
+    pub year: Option<YearInfo>,
+    pub month: Option<MonthInfo>,
+    pub day_of_month: Option<DayOfMonth>,
+    pub iso_weekday: Option<Weekday>,
```

Comment:
Should probably be named `weekday`. Anything else here that needs to be renamed?

Reply:
(obsolete; fields are private now)